### PR TITLE
Add public endpoint validation function

### PR DESCRIPTION
**Pull request by nooga**
This pull request potentially resolves issue #83.

By relocating the validation into a public function, it provides users with a way to prevent potential panic during Client initialization.

Before this change, users were required to conduct these checks independently and to keep these checks updated with each library upgrade. To simplify this process, a function is now available for users to call before setting up a new Client.

Additionally, this adjustment doesn't disrupt the API in a significant manner, nor does it impact existing client code.

### CODE REVIEW

**Comment by FZambia** [comment]
Hello,

Thanks for PR! 

I do not see why this should be in a library, if you want to check whether the endpoint is valid or not - you can do this outside centrifuge-go in your application code just before initializing Client. There are not so many rules for the endpoint string - it should be a valid URL with `ws` or `wss` scheme.

Probably client initialization should return an error at some point in the future, but I suppose the problem is pretty small here to introduce such a change right now.

What I think worth doing though looking at the current code is document which endpoint format Client initialization expects and that it will panic in case of unexpected WebSocket endpoint format.

**Comment by nooga** [comment]
Hi @FZambia,

I understand your point of view and I agree, it is not that big of a deal for library users to implement a pre-check before initializing the Client.

However, I'm coming from the maintainability perspective. In my view leaving the check up to the user makes it so that whenever you update the validation in the library all users should update their checks accordingly. Moreover, there would be no warning, so they just upgrade deps, code builds, tests and runs and all is good for a while.

Where is the risk? Let's say you update your validation to use proper URL parsing to find out if the URL is well-formed early. Myself, as your user will miss that and leave my pre-validation as it is. Now we have a situation where my check can pass and your check could fail with the same input. This would result in your lib panicking my long-running service at any random moment since deployment.

In anticipation of such scenario I would of course guard the Client initialization with a rescue but then my pre-check seems superfluous again. 

In short, my rationale behind the new function is that by having it provided by the lib I can always be in sync with your validation and be sure that Client init will never panic AND I don't have to rely on rescue as error handling mechanism.

It would be great if it just returned an error but I understand this kind of API change is not easy to do and will break client code so I'm not counting on this any time soon :)

Thanks!

**Comment by FZambia** [comment]
> Now we have a situation where my check can pass and your check could fail with the same input. This would result in your lib panicking my long-running service at any random moment since deployment.

That's why a proper documentation helps – to be explicit about library current expectations about an endpoint. Then if breaking changes happen - it will be reflected in doc or Client should start returning an error at some point. I can't imagine how endpoint may be additionally restricted inside centrifuge-go beyond being a valid URL and have ws/wss scheme.

Your case is quite special and it seems requires even more validation anyway than just looking at endpoint format. I already mentioned security aspect in the related issue. Another thing you can do is to write a unit test to check endpoints you accept from remote microservices can be used for initializing Client from centrifuge-go. You must be super-strict in what you accept from another microservices as an endpoint.

**Comment by nooga** [comment]
Thanks for your feedback and discussion! Closing this now.
